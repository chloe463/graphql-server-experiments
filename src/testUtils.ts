import { ApolloServer } from "apollo-server";
import { Context } from "./context";
import { schema } from "./graphql/schema";

type DummyContext = {
  [key in keyof Context]: jest.Mock;
};

export const constructTestServer = (context: DummyContext) => {
  const server = new ApolloServer({
    schema,
    context,
  });

  return server;
};
