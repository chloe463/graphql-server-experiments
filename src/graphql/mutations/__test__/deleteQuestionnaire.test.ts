import { PrismaClient } from "@prisma/client";
import gql from "graphql-tag";
import { constructTestServer } from "../../../testUtils";

const prismaClientMock = (PrismaClient as any) as jest.Mock<PrismaClient>;

type DeleteQuestionnaireVariables = {
  id: number;
}
type DeleteQuestionnaireResponse = {
  result: boolean;
};

const DELETE_QUESTIONNAIRE_MUTATION = gql`
  mutation DeleteQuestionnaire($id: Int!) {
    deleteQuestionnaire(id: $id) {
      result
    }
  }
`;

describe("[Mutation] deleteQuestionnaire", () => {
  it("can soft delete questionnaire", async () => {
    (prismaClientMock as any).questionnaire = {
      update: jest.fn().mockReturnValue(true),
    };
    const server = constructTestServer();

    const res = await server.executeOperation({
      query: DELETE_QUESTIONNAIRE_MUTATION,
      variables: {
        id: 1,
      },
    }, {
      contextValue: {
        prismaClient: prismaClientMock,
      }
    });
    expect(res).toMatchSnapshot();
  });
});
