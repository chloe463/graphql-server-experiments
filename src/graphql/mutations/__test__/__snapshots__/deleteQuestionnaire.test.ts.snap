// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`[Mutation] deleteQuestionnaire can soft delete questionnaire 1`] = `
{
  "body": {
    "kind": "single",
    "singleResult": {
      "data": {
        "deleteQuestionnaire": {
          "result": true,
        },
      },
      "errors": undefined,
    },
  },
  "http": {
    "headers": 
Map {
      "cache-control" => "no-store",
    }
,
    "status": undefined,
  },
}
`;
